
Starting with r223966 in 9-CURRENT, a new interface is available
(RFTSIGZMB).  At the time of writing glibc isn't using it.  Even
when it does, we need to keep this patch for backward
compatibility untill old versions of glibc are flushed out.

--- a/sys/kern/kern_fork.c
+++ b/sys/kern/kern_fork.c
@@ -480,7 +480,13 @@
 	if (flags & RFTSIGZMB)
 	        p2->p_sigparent = RFTSIGNUM(flags);
 	else if (flags & RFLINUXTHPN)
-	        p2->p_sigparent = SIGUSR1;
+	{
+		int sig;
+		sig = (flags >> 24) & 0x7F;
+	        if (sig == 0) sig = SIGUSR1;
+	        if (sig == SIGCHLD) sig = 0;
+	        p2->p_sigparent = sig;
+	}
 	else
 	        p2->p_sigparent = SIGCHLD;
 
